ra-optimisation/linear-rao/src/main/java/com/farao_community/farao/linear_rao/SystematicAnalysisEngine.java

```diff
@@ -166,13 +177,25 @@ private void checkSensiResults(LinearRaoData linearRaoData, SystematicSensitivit
      */
     private void setResults(LinearRaoData linearRaoData, SystematicSensitivityAnalysisResult systematicSensitivityAnalysisResult, Map<String, Double> loopflows) {
         linearRaoData.setSystematicSensitivityAnalysisResult(systematicSensitivityAnalysisResult);
+        //set functional cost
         double minMargin = getMinMargin(linearRaoData, systematicSensitivityAnalysisResult);
         linearRaoData.getCracResult().setFunctionalCost(-minMargin);
-        linearRaoData.getCracResult().setVirtualCost(fallbackMode ? linearRaoParameters.getFallbackOvercost() : 0);
+
+        //update virtual cost
+        double virtualCost = linearRaoData.getCracResult().getVirtualCost();
+        double fallbackModeVirtualCost = fallbackMode ? linearRaoParameters.getFallbackOvercost() : 0.0;
+        linearRaoData.getCracResult().setVirtualCost(virtualCost + fallbackModeVirtualCost);
+
+        linearRaoData.getCracResult().setNetworkSecurityStatus(minMargin < 0 ? CracResult.NetworkSecurityStatus.UNSECURED : CracResult.NetworkSecurityStatus.SECURED);
         if (isLoopflowViolation()) {
-//            linearRaoData.getCracResult().setCost(Double.POSITIVE_INFINITY); //todo: set a high cost if loopflow constraint violation => use "virtual cost" in the future
+            if (linearRaoParameters.getExtendable().getLoopflowViolationCost() == 0.0) {
+                linearRaoData.getCracResult().setNetworkSecurityStatus(CracResult.NetworkSecurityStatus.UNSECURED); //flag UNSECURED if loopflowViolation and no virtual cost
+                linearRaoData.getCracResult().setVirtualCost(DEFAULT_LOOPFLOWVIOLATION_VIRTUALCOST); // "zero-loopflowViolationCost", no virtual cost available from Linear optim, set to MAX
+            } else {
+                LOGGER.info("Loopflow violation cost is positive and loopflow constraint violation");
+            }
```

Comment:
I'm not sure that the violation cost is set here if it is != 0

Reply:
virtual cost from solver is set in fillVirtualCostInCracResult

Reply:
Hmm.. I see, but it is made just after the linear optimization, with the flow estimated by the linear optimization (approximated) and not with the flow in output of the sensi.

Reply:
Yes

Reply:
It does not matter. If the virtual variable is not zero, it will has a high virtual cost, otherwise it is zero.